src/backend/core/authentication/backends.py

```diff
@@ -105,3 +105,32 @@ def create_user(self, claims):
             email=claims.get("email"),
             name=claims.get("name"),
         )
+
+    def compute_full_name(self, user_info):
+        """Compute user's full name based on OIDC fields in settings."""
+        name_fields = settings.USER_OIDC_FIELDS_TO_NAME
+        full_name = " ".join(
+            user_info[field] for field in name_fields if user_info.get(field)
+        )
+        return full_name or None
+
+    def get_existing_user(self, sub, email):
+        """Fetch existing user by sub or email."""
+        try:
+            return User.objects.get(sub=sub, is_active=True)
```

Comment:
We should not check the `is_active` here, because the result will be a duplicated user with the same `sub`. We should get the user, then later in the calling function, check the `is_active`.

```suggestion
            return User.objects.get(sub=sub)
```